src/components/NewSnippet.jsx

```diff
@@ -1,49 +1,87 @@
 import React from 'react';
+import { connect } from 'react-redux';
 import { Controlled as CodeMirror } from 'react-codemirror2';
 
 import 'codemirror/lib/codemirror.css';
 
 import Title from './common/Title';
 import Input from './common/Input';
 import Syntaxes from './Syntaxes';
+import * as actions from '../actions';
 
 import '../styles/NewSnippet.styl';
 
 class NewSnippet extends React.Component {
-  constructor() {
-    super();
-    this.state = { code: '' };
+  constructor(props) {
+    super(props);
+    this.state = {
+      content: '',
+      title: '',
+      tags: [],
+      syntax: '', // eslint-disable-line react/no-unused-state
+    };
+    this.postSnippet = this.postSnippet.bind(this);
+    this.onSyntaxClick = this.onSyntaxClick.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
   }
+
+  onSyntaxClick(syntax) {
+    this.setState({ syntax }); // eslint-disable-line react/no-unused-state
+  }
+
+  onInputChange(e) {
+    const { name, value } = e.target;
+    const result = (name === 'tags')
+      ? value.split(',').map(item => item.trim())
+      : value;
+
+    this.setState({ [name]: result });
+  }
+
+  postSnippet(e) {
+    e.preventDefault();
```

Comment:
I'd add few lines with explanations why `preventDefault()` is necessary. If I'm not mistaken it's because we don't want to follow default HTML behavior when form data is submitted to specified URI using `POST` and `multipart/form-data` content type.

Reply:
hm.It seems obvious to me